### PR TITLE
msgpack-cxx: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **msgpack-cxx/4.0.3**

This recipe is separated from msgpack.
Because msgpack is separated C and C++ code base.

See also https://github.com/conan-io/conan-center-index/pull/8183.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
So the plan is to deprecate the existing `msgpack` recipe right ?

**Comment by toge** [comment]
@ericLemanissier 
Yes. I'm going to do so after this PR is merged.

File: recipes/msgpack-cxx/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        self.options["boost"].header_only = False
+        self.options["boost"].without_chrono = False
+        self.options["boost"].without_context = False
+        self.options["boost"].without_system = False
+        self.options["boost"].without_timer = False
```
Comment (ericLemanissier) [inline recipes/msgpack-cxx/all/conanfile.py:30]: If this is a requirement, please raise an exception in `validate(self)` if any of these options have incorrect values (in addition of setting the values in `configure(self):`

File: recipes/msgpack-cxx/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(msgpack-cxx REQUIRED CONFIG)
+
+  add_executable(${PROJECT_NAME} test_package.cpp)
+  target_link_libraries(${PROJECT_NAME} msgpack::msgpack-cxx)
+  set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```
Comment (ericLemanissier) [inline recipes/msgpack-cxx/all/test_package/CMakeLists.txt:11]:
```suggestion
add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} msgpack::msgpack-cxx)
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

File: recipes/msgpack-cxx/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+required_conan_version = ">=1.36.0"
+
+class MsgpackCXXConan(ConanFile):
+    name = "msgpack-cxx"
+    description = "The official C++ library for MessagePack"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/msgpack/msgpack-c"
+    topics = ("msgpack", "message-pack", "serialization")
+    license = "BSL-1.0"
+    generators = "cmake"
```
Comment (ericLemanissier) [inline recipes/msgpack-cxx/all/conanfile.py:14]:
```suggestion
```

File: recipes/msgpack-cxx/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/msgpack-cxx/all/conanfile.py]: Fixed

**Review by ericLemanissier — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by SpaceIm** [comment]
The split in 2 recipes is quite problematic. Upstream, both libs share the same config file, and it's not possible to achieve that in conan with 2 recipes, only with the mono recipe.

**Pull request merged**